test/test_python_dispatch.py

```diff
@@ -12,6 +12,86 @@
 import logging
 from functools import partial
 
+class TestPythonRegistration(TestCase):
+    def test_override_aten_ops_with_multiple_libraries(self) -> None:
+        x = torch.tensor([1, 2])
+        my_lib1 = torch.library.extend_library("aten")
+        my_lib2 = torch.library.extend_library("aten")
+
+        # Example 1
+        def my_neg(*args, **kwargs):
+            return args[0]._neg_view()
+
+        # Now we are secretly making the operator a view op so autograd needs to know how
+        # to handle it
+        my_lib1.impl('neg', my_neg, "AutogradCPU")
+
+        self.assertTrue(torch.neg(x).is_neg())
+
+        # RuntimeError: impl("aten::neg", ...):
+        # Explicitly provided namespace (aten) in operator name does not match ...
+        with self.assertRaisesRegex(RuntimeError, "operator name does not match namespace"):
+            my_lib3 = torch.library.extend_library("foo")
+            my_lib3.impl(torch.ops.aten.neg.default, my_neg, "AutogradCPU")
+            my_lib3.remove()
+
+        # Example 2
+        def my_mul(*args, **kwargs):
+            return torch.zeros_like(args[0])
+
+        # torch.ops.aten.mul.Tensor
+        my_lib2.impl("aten::mul.Tensor", my_mul, "ZeroTensor")
+
+        y = torch._efficientzerotensor(2)
+        self.assertFalse(torch.mul(x, y)._is_zerotensor())
+
+        # Assert that a user can't override the behavior of a (ns, op, dispatch_key)
+        # combination if someone overrided the behavior for the same before them
+        with self.assertRaisesRegex(RuntimeError, 'already a kernel registered from python'):
+            my_lib2.impl(torch.ops.aten.mul.Tensor, my_mul, "ZeroTensor")
+
+        my_lib1.remove()
+
+        # Validate that lib2 is not affected by removing lib1
+        self.assertFalse(torch.mul(x, y)._is_zerotensor())
+
+        my_lib2.remove()
+
+        # Validate that the old behavior is restored for neg and mul
+        self.assertFalse(torch.neg(x).is_neg())
+        self.assertTrue(torch.mul(x, y)._is_zerotensor())
+
+    def test_override_cpu_sum(self) -> None:
+        # Example 1
+        run = [False]
+
+        def my_sum(*args, **kwargs):
+            run[0] = True
+            return args[0]
+
+        my_lib1 = torch.library.extend_library("aten")
+        my_lib1.impl('aten::sum', my_sum, "CPU")
```

Comment:
We really need to specify what kind of string is allowed for `op_name` here. It seems that both `sum` and `aten::sum` works based on the tests? But if I pass `foo::sum`, will that work?

Reply:
If foo is a valid namespace, then it should work

Reply:
Even though the library was created for the namespace "aten"?